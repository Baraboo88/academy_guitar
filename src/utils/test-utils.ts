import {AddCommentModel, GuitarCommentModel, GuitarModel, GuitarType} from '../types/guitar-model';
import {createApi} from '../api';
import {applyMiddleware, createStore} from 'redux';
import thunk from 'redux-thunk';
import {ReactWrapper} from 'enzyme';


export const MOCK_ERROR= 'Somme error';

export const mockGuitars: GuitarModel[] = [
  {
    id: 1,
    name: 'Честер Bass',
    vendorCode: 'SO757575',
    type: 'electric' as GuitarType,
    description: 'Замечательный малобюджетный вариант, созданный для новичков, которые отдают предпочтение мелодичным стилям. Прекрасно звучат блюз и баллады, исполненные на этой гитаре. Акустические свойства весьма высоки, в отличие от ее стоимости.',
    previewImg: 'img/guitar-1.jpg',
    stringCount: 7,
    rating: 4,
    price: 17500,
  },
  {
    id: 2,
    name: 'CURT Z300',
    vendorCode: 'TK129049',
    type: 'electric' as GuitarType,
    description: 'Эргономичность гитары и качество сборки являются, пожалуй, её главными преимуществами. Идеальное расположение в руках музыканта дополняется прочностью конструкции из клёна.',
    previewImg: 'img/guitar-8.jpg',
    stringCount: 7,
    rating: 3.5,
    price: 29500,
  },
  {
    id: 3,
    name: 'Roman LX',
    vendorCode: 'RO111111',
    type: 'ukulele' as GuitarType,
    description: 'Укулеле класса премиум от компании CURT, собравшая в себе все самые необходимые качесва: лёгкость корпуса, прочность струн и компактный размер.',
    previewImg: 'img/guitar-6.jpg',
    stringCount: 4,
    rating: 4,
    price: 6800,
  },
  {
    id: 4,
    name: 'CURT T300',
    vendorCode: 'TK436457',
    type: 'electric' as GuitarType,
    description: 'CURT T300 - это шестиструнная электрогитара популярной линейки FPT. Модель с классическим стилем головы грифа и деки.',
    previewImg: 'img/guitar-3.jpg',
    stringCount: 6,
    rating: 5,
    price: 30000,
  },
  {
    id: 5,
    name: 'Dania Super',
    vendorCode: 'DI192138',
    type: 'acoustic' as GuitarType,
    description: 'Гитары производителя Dania пользуются популярностью у музыкантов разного уровня. В модели Super идеально сочетаются демократичная цена и качество. Корпус, выполненный из тополя позволяет корпусу долгое время сохранять свой первоначальный вид.',
    previewImg: 'img/guitar-4.jpg',
    stringCount: 7,
    rating: 4.5,
    price: 3500,
  },
  {
    id: 6,
    name: 'Честер WX',
    vendorCode: 'SO934345',
    type: 'electric' as GuitarType,
    description: 'Электрогитара с олд-скульным грифом и глянцевым корпусом для настоящих рок-музыкантов. Насыщенный звук идеально подойдет для исполнения рок-композиций.',
    previewImg: 'img/guitar-2.jpg',
    stringCount: 6,
    rating: 3.5,
    price: 15300,
  },
  {
    id: 7,
    name: 'Dania VX',
    vendorCode: 'DI082347',
    type: 'ukulele' as GuitarType,
    description: 'Укулеле - прекрасный компактный аналог классической акустической гитары. Прекрасное качество продуктов компании Dania сочетается с демократичными ценами продуктов линейки GSR.',
    previewImg: 'img/guitar-5.jpg',
    stringCount: 4,
    rating: 2.5,
    price: 2200,
  },
];
export const testInitialState = {
  currentGuitar: {
    currentGuitar: mockGuitars,
    errorMsg: '',
  },
  guitars: {
    guitars: mockGuitars,
    errorMsg: '',
  },
};

export const mockCommentToAdd: AddCommentModel = {
  advantage: 'Mock Advantage', comment: 'Mock Comment', disadvantage: 'Mock Disadvantage', guitarId: 3, rating: 3, userName: 'Some name',
};


export const mockComment: GuitarCommentModel = {id: '31b10247-adff-44a7-937e-c71af64f425b',
  userName: 'Никита',
  advantage: 'Цена.',
  disadvantage: 'Быстро приходит в негодность покрытие краской.',
  comment: 'Хорошая гитара для начинающих, сделана качественно. Лучше за эти деньги не найти.',
  rating: 5,
  createAt: new Date('2022-03-10T10:56:01.590Z'),
  guitarId: 1};

export const getTestStore = () => {

  const reducer = (state = testInitialState) => state;
  const api = createApi();

  return createStore(reducer, applyMiddleware(thunk.withExtraArgument(api)));
};

export const findByTestAtr = (component:ReactWrapper, attr: string) => component.find(`[data-test='${attr}']`);
