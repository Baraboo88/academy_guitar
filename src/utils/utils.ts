const STARTS_AMOUNT = 5;

export const imageAdapter = (img: string) => img.slice(img.length - 5, img.length - 4);


export const generateStarsArray = () => {
  const stars = [];
  for (let i = 1; i <= STARTS_AMOUNT; i++) {
    stars.push(i);
  }
  return stars;
};

export const getCyrillicRating = (rating: number) => {
  if(rating> 0 && rating < 3){
    return 'Плохо';
  } else if (rating >= 3 && rating < 4){
    return 'Удовлетворительно';
  } else if(rating >= 4 && rating < 5){
    return 'Хорошо';
  } else {
    return 'Отлично';
  }
};
