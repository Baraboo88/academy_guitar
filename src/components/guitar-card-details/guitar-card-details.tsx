import React, {useCallback, useEffect, useRef, useState} from 'react';
import {StateModel} from '../../types/redux-models';
import {getCurrentGuitar, getCurrentGuitarError, getCurrentGuitarIsResponseReceived} from '../../store/current-guitar/current-guitar-selectors';
import {connect} from 'react-redux';
import {AddCommentModel, GuitarCommentModel, GuitarModel} from '../../types/guitar-model';
import {Link, useNavigate, useParams} from 'react-router-dom';
import {getCyrillicRating, getCyrillicType, getPriceWithSpaces, getAdapterImage, ErrorMsg} from '../../utils/utils';
import {renderStars, StarSize} from '../guitar-card/guitar-card';
import Header from '../header/header';
import Footer from '../footer/footer';
import CommentCard from '../comment-card/comment-card';
import AddCommentModal from '../add-comment-modal/add-comment-modal';
import AddCommentModalSuccess from '../add-comment-modal-success/add-comment-modal-success';
import AddToCartModal from '../add-to-cart-modal/add-to-cart-modal';
import useOnScreen from '../../hooks/use-on-screen/use-on-screen';
import {CurrentGuitarActionCreator, CurrentGuitarOperation} from '../../store/current-guitar/current-guitar-reducer';


const COMMENTS_TO_SKIP = 3;
const DELAY_MIL_SEC = 1000;
interface GuitarCardDetailsProps {
    currentGuitar: GuitarModel | null;
    resetCurrentGuitar: () => void;
    onMount: (id: number) => void;
    getComments: (guitar: GuitarModel) => void;
    addComment: (guitar: GuitarModel, comment: AddCommentModel) => void;
    isResponseReceived: boolean;
    resetIsResponseReceived: () => void;
    error: string;
}

export enum ActiveTab{
  Characteristics= 'characteristics',
  Description='description'
}

function GuitarCardDetails(props: GuitarCardDetailsProps ) {

  const {currentGuitar, resetCurrentGuitar, onMount, getComments, addComment, isResponseReceived, error, resetIsResponseReceived} = props;
  const [commentsToSkip, setCommentsToSkip] = useState(COMMENTS_TO_SKIP);
  const [isAddCommentOpened, setIsAddCommentOpened] = useState(false);

  const [userName, setUserName] = useState<string>('');
  const [advantage, setAdvantage] = useState<string>('');
  const [disadvantage, setDisadvantage] = useState<string>('');
  const [comment, setComment] = useState<string>('');
  const [rating, setRating] = useState<number>(0);
  const [modalCommentIsSending, setModalCommentIsSending] = useState(false);
  const [isCommentAddSuccess, setIsCommentAddSuccess] = useState(false);

  //validation
  const [isNameValidationError, setIsNameValidationError] = useState(false);
  const [isAdvantageValidationError, setIsAdvantageValidationError] = useState(false);
  const [isDisadvantageValidationError, setIsDisadvantageValidationError] = useState(false);
  const [isCommentValidationError, setIsCommentValidationError] = useState(false);
  const [isRatingValidationError, setIsRatingValidationError] = useState(false);

  const [activeTab, setActiveTab] = useState<ActiveTab>();
  const [isAddToCardPopUpOpened, setIsAddToCardPopUpOpened] = useState(false);
  const [isCommentsLoading, setIsCommentsLoading] = useState(false);

  const ref = useRef<HTMLDivElement>(null);
  const isVisible = useOnScreen(ref);
  const navigate = useNavigate();

  const {id, cat} = useParams();

  const handlerCommentsShowMore = useCallback(() => {

    setCommentsToSkip(commentsToSkip + COMMENTS_TO_SKIP);
    setIsCommentsLoading(true);
  },[commentsToSkip, setIsCommentsLoading]);

  useEffect(() => {
    if(isVisible && commentsToSkip !== COMMENTS_TO_SKIP){

      if(!isCommentsLoading){

        handlerCommentsShowMore();
      }
      setTimeout(() => {
        setIsCommentsLoading(false);
      }, DELAY_MIL_SEC);
    }
  },[isVisible, handlerCommentsShowMore, commentsToSkip, isCommentsLoading, setIsCommentsLoading]);

  useEffect(() => {
    if (error === ErrorMsg.NotFound) {
      navigate('/not-found');
    }
  }, [error, navigate]);

  useEffect( () => {
    onMount(Number(id));

    return () => {
      resetCurrentGuitar();
    };

  }, [id, onMount, resetCurrentGuitar]);

  useEffect(() => {
    if(currentGuitar) {
      if( cat === ActiveTab.Characteristics){
        setActiveTab(ActiveTab.Characteristics);
      }

      if( cat === ActiveTab.Description){
        setActiveTab(ActiveTab.Description);
      }
    } else {
      setActiveTab(ActiveTab.Characteristics);
    }

  }, [cat, currentGuitar]);

  useEffect(() => {

    if (currentGuitar && !currentGuitar.comments) {
      getComments(currentGuitar);

    }

  }, [currentGuitar, getComments]);

  useEffect(() => {

    if(modalCommentIsSending && isResponseReceived && !error){
      setModalCommentIsSending(false);
      setIsCommentAddSuccess(true);
      setUserName('');
      setRating(0);
      setAdvantage('');
      setDisadvantage('');
      setComment('');
    }
  },[modalCommentIsSending, isResponseReceived, error]);

  useEffect(() => () => {
    resetIsResponseReceived();
  },[resetIsResponseReceived]);

  const handlerCommentSubmit = (commentAdded: AddCommentModel) => {
    if(!commentAdded.userName){
      setIsNameValidationError(true);
    }
    if(!commentAdded.advantage){
      setIsAdvantageValidationError(true);
    }
    if(!commentAdded.disadvantage){
      setIsDisadvantageValidationError(true);
    }
    if(!commentAdded.comment){
      setIsCommentValidationError(true);
    }
    if(!commentAdded.rating){
      setIsRatingValidationError(true);
    }
    if(commentAdded.userName && commentAdded.advantage && commentAdded.disadvantage && commentAdded.comment && commentAdded.rating&& currentGuitar){
      addComment(currentGuitar,commentAdded);
      setModalCommentIsSending(true);
    }
  };

  const handlerAddToCartModalClose = () => {
    setIsAddToCardPopUpOpened(false);
  };

  const handlerUserNameSet = (name: string) => {
    setUserName(name);
    setIsNameValidationError(false);
  };

  const handlerAdvantageSet = (adv: string) => {
    setAdvantage(adv);
    setIsAdvantageValidationError(false);
  };

  const handlerDisadvantageSet = (disadv: string) => {
    setDisadvantage(disadv);
    setIsDisadvantageValidationError(false);
  };

  const handlerCommentSet = (userComment: string) => {
    setComment(userComment);
    setIsCommentValidationError(false);
  };

  const handlerRatingSet = (userRating: number) => {
    setRating(userRating);
    setIsRatingValidationError(false);
  };

  const handlerAddCommentModalClose = () => {
    setIsAddCommentOpened(false);
  };


  return (
    <div className="wrapper">

      <Header/>
      <main className="page-content">
        {currentGuitar &&
            <div className="container">
              <h1 className="page-content__title title title--bigger">{currentGuitar.name}</h1>
              <ul className="breadcrumbs page-content__breadcrumbs">
                <li className="breadcrumbs__item"><Link to={'/'} className="link">Главная</Link>
                </li>
                <li className="breadcrumbs__item"><Link to={'/'} className="link">Каталог</Link>
                </li>
                <li className="breadcrumbs__item"><span >{currentGuitar.name}</span>
                </li>
              </ul>


              <div className="product-container">
                <img className="product-container__img"
                  src={`/img/content/catalog-product-${getAdapterImage(currentGuitar.previewImg)}.jpg`}
                  srcSet={`/img/content/catalog-product-${getAdapterImage(currentGuitar.previewImg)}@2x.jpg 2x`}
                  width="90"
                  height="235" alt=""
                />
                <div className="product-container__info-wrapper">
                  <h2 className="product-container__title title title--big title--uppercase">{currentGuitar.name}</h2>
                  <div className="rate product-container__rating">
                    {renderStars(currentGuitar.rating, StarSize.CardDetails)}
                    <p className="visually-hidden">Оценка: {getCyrillicRating(currentGuitar.rating)}</p>
                    <p className="rate__count">
                      <span
                        className="visually-hidden"
                      >Всего оценок:
                      </span> {currentGuitar.comments ? currentGuitar.comments.length : 0}
                    </p>
                  </div>
                  <div className="tabs">
                    <Link to={`/product/${id}/${ActiveTab.Characteristics}`}
                      className={`button button--medium tabs__button ${activeTab !== ActiveTab.Characteristics ? 'button--black-border' : ''}`}
                    >
                      Характеристики
                    </Link>
                    <Link
                      to={`/product/${id}/${ActiveTab.Description}`}
                      className={`button button--medium tabs__button ${activeTab !== ActiveTab.Description ? 'button--black-border' : ''} `}
                    >Описание
                    </Link>
                    <div className="tabs__content" id="characteristics">
                      <table className={`tabs__table ${activeTab !== ActiveTab.Characteristics ? 'hidden' : ''}`}>
                        <tbody>
                          <tr className="tabs__table-row">
                            <td className="tabs__title">Артикул:</td>
                            <td className="tabs__value">{currentGuitar.vendorCode}</td>
                          </tr>
                          <tr className="tabs__table-row">
                            <td className="tabs__title">Тип:</td>
                            <td className="tabs__value">{getCyrillicType(currentGuitar.type)}</td>
                          </tr>
                          <tr className="tabs__table-row">
                            <td className="tabs__title">Количество струн:</td>
                            <td className="tabs__value">{currentGuitar.stringCount} струнная</td>
                          </tr>
                        </tbody>
                      </table>
                      <p className={`tabs__product-description  ${activeTab !== ActiveTab.Description ? 'hidden' : ''}`}>{currentGuitar.description}
                      </p>
                    </div>
                  </div>
                </div>
                <div className="product-container__price-wrapper">
                  <p className="product-container__price-info product-container__price-info--title">Цена:</p>
                  <p className="product-container__price-info product-container__price-info--value">{getPriceWithSpaces(currentGuitar.price)} ₽</p>
                  <button
                    onClick={() => {
                      setIsAddToCardPopUpOpened(true);
                    }}
                    className="button button--red button--big product-container__button"
                  >Добавить в
                    корзину
                  </button>
                </div>
              </div>

              <section className="reviews">
                <h3 className="reviews__title title title--bigger">Отзывы</h3>
                <button onClick={() => {
                  setIsAddCommentOpened(true);
                  setIsCommentAddSuccess(false);
                }}
                className="button button--red-border button--big reviews__sumbit-button"
                >Оставить отзыв
                </button>

                {(currentGuitar?.comments && currentGuitar?.comments.length > 0) ? currentGuitar?.comments.slice(0, commentsToSkip).map((innerComment:GuitarCommentModel) =>
                  <CommentCard key={innerComment.id} comment={innerComment}/>) : <div/>}

                {(currentGuitar?.comments && commentsToSkip < currentGuitar?.comments?.length  && currentGuitar?.comments.length > 0 && commentsToSkip < currentGuitar?.comments.length) &&

                    <button onClick={() => {
                      handlerCommentsShowMore();
                    }} className="button button--medium reviews__more-button"
                    >Показать еще отзывы
                    </button>}


                <button className="button button--up button--red-border button--big reviews__up-button"
                  onClick={() => {
                    window.scrollTo(0, 0);
                  }}
                >Наверх
                </button>

              </section>
              {isAddCommentOpened && (isCommentAddSuccess ?
                <AddCommentModalSuccess onModalClose={handlerAddCommentModalClose}/> :
                <AddCommentModal onCloseModal={handlerAddCommentModalClose} guitar={currentGuitar}
                  userName={userName} onSetUserName={handlerUserNameSet}
                  advantage={advantage} onSetAdvantage={handlerAdvantageSet}
                  disadvantage={disadvantage} onSetDisadvantage={handlerDisadvantageSet}
                  rating={rating} onSetRating={handlerRatingSet}
                  comment={comment} onSetComment={handlerCommentSet}
                  onSubmitHandler={handlerCommentSubmit}
                  isNameValidationError ={isNameValidationError}
                  isAdvantageValidationError = {isAdvantageValidationError}
                  isDisadvantageValidationError ={isDisadvantageValidationError}
                  isCommentValidationError={isCommentValidationError}
                  isRatingValidationError={isRatingValidationError}
                />)}

              {isAddToCardPopUpOpened && <AddToCartModal onModalClose={handlerAddToCartModalClose} guitar={currentGuitar}/>}
            </div>}
        <div  ref={ref}></div>
      </main>
      <Footer />
    </div>
  );
}

const mapStateToProps = (state: StateModel) => ({
  currentGuitar: getCurrentGuitar(state),
  isResponseReceived: getCurrentGuitarIsResponseReceived(state),
  error: getCurrentGuitarError(state),
});

/* eslint-disable  @typescript-eslint/no-explicit-any */
const mapDispatchToProps = (dispatch: any) => (
  {
    onMount(id: number) {
      dispatch(CurrentGuitarOperation.getGuitarById(id));
    },
    getComments(guitar: GuitarModel) {
      dispatch(CurrentGuitarOperation.getCurrentGuitarComments(guitar));
    },
    resetCurrentGuitar() {
      dispatch(CurrentGuitarActionCreator.setCurrentGuitar(null));
      dispatch(CurrentGuitarActionCreator.setIsResponseReceived(false));
      dispatch(CurrentGuitarActionCreator.setError(''));
    },
    addComment(guitar: GuitarModel, comment: AddCommentModel){
      dispatch(CurrentGuitarOperation.addComment(guitar, comment));
    },
    resetIsResponseReceived() {
      dispatch(CurrentGuitarActionCreator.setIsResponseReceived(false));
    },
  });

export default connect(mapStateToProps, mapDispatchToProps)(GuitarCardDetails);
export {GuitarCardDetails};
