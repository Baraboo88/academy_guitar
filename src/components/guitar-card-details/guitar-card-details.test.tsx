
import * as Enzyme from 'enzyme';

import { mount } from 'enzyme';
import EnzymeReactAdapter from '@wojtekmaj/enzyme-adapter-react-17';

import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';

import {getTestStore, mockGuitars} from '../../utils/test-utils';
import {ActiveTab, GuitarCardDetails} from './guitar-card-details';
import { BrowserRouter } from 'react-router-dom';


Enzyme.configure({adapter: new EnzymeReactAdapter()});


it('GuitarCardDetails successfully rendered', () => {
  const mockResetCurrentGuitar = jest.fn();
  const mockOnMount = jest.fn();
  const mockGetComments = jest.fn();
  const mockAddComment = jest.fn();
  const mockResetIsResponseReceived = jest.fn();
  const MOCK_ERROR = '';

  const mockHistory = { push: jest.fn() };
  const routeComponentPropsMock = {
    history: mockHistory as any,
    location: {} as any,
    match: {
      params: {
        id: '1',
        cat: ActiveTab.Characteristics,
      },
    } as any,
  };

  const tree = mount(
    <Provider store={getTestStore()}>
      <BrowserRouter>
        <GuitarCardDetails
          {...routeComponentPropsMock}
          currentGuitar={mockGuitars[0]}
          resetCurrentGuitar ={mockResetCurrentGuitar}
          onMount = {mockOnMount}
          getComments={mockGetComments}
          addComment={mockAddComment}
          isResponseReceived ={false}
          resetIsResponseReceived={mockResetIsResponseReceived}
          error={MOCK_ERROR}
        />
      </BrowserRouter>
    </Provider>,
  );
  expect(toJson(tree, {mode: 'deep'})).toMatchSnapshot();
});
