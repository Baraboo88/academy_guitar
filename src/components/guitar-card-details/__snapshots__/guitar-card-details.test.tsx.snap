// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuitarCardDetails successfully rendered 1`] = `
<div
  className="wrapper"
>
  <header
    className="header"
    id="header"
  >
    <div
      className="container header__wrapper"
    >
      <a
        className="header__logo logo"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="Логотип"
          className="logo__img"
          height="70"
          src="/img/svg/logo.svg"
          width="70"
        />
      </a>
      <nav
        className="main-nav"
      >
        <ul
          className="main-nav__list"
        >
          <li>
            <a
              className="link main-nav__link link--current"
              href="#"
            >
              Каталог
            </a>
          </li>
          <li>
            <a
              className="link main-nav__link"
              href="#"
            >
              Где купить?
            </a>
          </li>
          <li>
            <a
              className="link main-nav__link"
              href="#"
            >
              О компании
            </a>
          </li>
        </ul>
      </nav>
      <div
        className="form-search"
      >
        <form
          className="form-search__form"
          id="form-search"
        >
          <button
            className="form-search__submit"
            type="submit"
          >
            <svg
              aria-hidden="true"
              className="form-search__icon"
              height="15"
              width="14"
            >
              <use
                xlinkHref="#icon-search"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              Начать поиск
            </span>
          </button>
          <input
            autoComplete="off"
            className="form-search__input"
            id="search"
            placeholder="что вы ищите?"
            type="text"
          />
          <label
            className="visually-hidden"
            htmlFor="search"
          >
            Поиск
          </label>
        </form>
        <ul
          className="form-search__select-list hidden"
        >
          <li
            className="form-search__select-item"
            tabIndex={0}
          >
            Четстер Plus
          </li>
          <li
            className="form-search__select-item"
            tabIndex={0}
          >
            Четстер UX
          </li>
          <li
            className="form-search__select-item"
            tabIndex={0}
          >
            Четстер UX2
          </li>
          <li
            className="form-search__select-item"
            tabIndex={0}
          >
            Четстер UX3
          </li>
          <li
            className="form-search__select-item"
            tabIndex={0}
          >
            Четстер UX4
          </li>
          <li
            className="form-search__select-item"
            tabIndex={0}
          >
            Четстер UX5
          </li>
        </ul>
        <button
          className="form-search__reset"
          form="form-search"
          type="reset"
        >
          <svg
            aria-hidden="true"
            className="form-search__icon"
            height="15"
            width="14"
          >
            <use
              xlinkHref="#icon-close"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            Сбросить поиск
          </span>
        </button>
      </div>
      <a
        aria-label="Корзина"
        className="header__cart-link"
        href="#"
      >
        <svg
          aria-hidden="true"
          className="header__cart-icon"
          height="14"
          width="14"
        >
          <use
            xlinkHref="#icon-basket"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          Перейти в корзину
        </span>
        <span
          className="header__cart-count"
        >
          2
        </span>
      </a>
    </div>
  </header>
  <main
    className="page-content"
  >
    <div
      className="container"
    >
      <h1
        className="page-content__title title title--bigger"
      >
        Честер Bass
      </h1>
      <ul
        className="breadcrumbs page-content__breadcrumbs"
      >
        <li
          className="breadcrumbs__item"
        >
          <a
            className="link"
            href="/"
            onClick={[Function]}
          >
            Главная
          </a>
        </li>
        <li
          className="breadcrumbs__item"
        >
          <a
            className="link"
            href="/"
            onClick={[Function]}
          >
            Каталог
          </a>
        </li>
        <li
          className="breadcrumbs__item"
        >
          <span>
            Честер Bass
          </span>
        </li>
      </ul>
      <div
        className="product-container"
      >
        <img
          alt=""
          className="product-container__img"
          height="235"
          src="/img/content/catalog-product-1.jpg"
          srcSet="/img/content/catalog-product-1@2x.jpg 2x"
          width="90"
        />
        <div
          className="product-container__info-wrapper"
        >
          <h2
            className="product-container__title title title--big title--uppercase"
          >
            Честер Bass
          </h2>
          <div
            className="rate product-container__rating"
          >
            <svg
              aria-hidden="true"
              height="14"
              key="4 - 1"
              width="14"
            >
              <use
                xlinkHref="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="14"
              key="4 - 2"
              width="14"
            >
              <use
                xlinkHref="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="14"
              key="4 - 3"
              width="14"
            >
              <use
                xlinkHref="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="14"
              key="4 - 4"
              width="14"
            >
              <use
                xlinkHref="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="14"
              key="4 - 5"
              width="14"
            >
              <use
                xlinkHref="#icon-star"
              />
            </svg>
            <p
              className="visually-hidden"
            >
              Оценка: 
              Хорошо
            </p>
            <p
              className="rate__count"
            >
              <span
                className="visually-hidden"
              >
                Всего оценок:
              </span>
               
              0
            </p>
          </div>
          <div
            className="tabs"
          >
            <a
              className="button button--medium tabs__button button--black-border"
              href="/product/undefined/characteristics"
              onClick={[Function]}
            >
              Характеристики
            </a>
            <a
              className="button button--medium tabs__button button--black-border "
              href="/product/undefined/description"
              onClick={[Function]}
            >
              Описание
            </a>
            <div
              className="tabs__content"
              id="characteristics"
            >
              <table
                className="tabs__table hidden"
              >
                <tbody>
                  <tr
                    className="tabs__table-row"
                  >
                    <td
                      className="tabs__title"
                    >
                      Артикул:
                    </td>
                    <td
                      className="tabs__value"
                    >
                      SO757575
                    </td>
                  </tr>
                  <tr
                    className="tabs__table-row"
                  >
                    <td
                      className="tabs__title"
                    >
                      Тип:
                    </td>
                    <td
                      className="tabs__value"
                    >
                      Электрогитара
                    </td>
                  </tr>
                  <tr
                    className="tabs__table-row"
                  >
                    <td
                      className="tabs__title"
                    >
                      Количество струн:
                    </td>
                    <td
                      className="tabs__value"
                    >
                      7
                       струнная
                    </td>
                  </tr>
                </tbody>
              </table>
              <p
                className="tabs__product-description  hidden"
              >
                Замечательный малобюджетный вариант, созданный для новичков, которые отдают предпочтение мелодичным стилям. Прекрасно звучат блюз и баллады, исполненные на этой гитаре. Акустические свойства весьма высоки, в отличие от ее стоимости.
              </p>
            </div>
          </div>
        </div>
        <div
          className="product-container__price-wrapper"
        >
          <p
            className="product-container__price-info product-container__price-info--title"
          >
            Цена:
          </p>
          <p
            className="product-container__price-info product-container__price-info--value"
          >
            17 500
             ₽
          </p>
          <button
            className="button button--red button--big product-container__button"
            onClick={[Function]}
          >
            Добавить в корзину
          </button>
        </div>
      </div>
      <section
        className="reviews"
      >
        <h3
          className="reviews__title title title--bigger"
        >
          Отзывы
        </h3>
        <button
          className="button button--red-border button--big reviews__sumbit-button"
          onClick={[Function]}
        >
          Оставить отзыв
        </button>
        <div />
        <button
          className="button button--up button--red-border button--big reviews__up-button"
          onClick={[Function]}
        >
          Наверх
        </button>
      </section>
    </div>
    <div />
  </main>
  <footer
    className="footer"
  >
    <div
      className="footer__container container"
    >
      <a
        className="footer__logo logo"
        href="/"
      >
        <img
          alt="Логотип"
          className="logo__img"
          height="70"
          src="./img/svg/logo.svg"
          width="70"
        />
      </a>
      <div
        className="socials footer__socials"
      >
        <ul
          className="socials__list"
        >
          <li
            className="socials-item"
          >
            <a
              aria-label="skype"
              className="socials__link"
              href="https://www.skype.com/"
            >
              <svg
                aria-hidden="true"
                className="socials__icon"
                height="24"
                width="24"
              >
                <use
                  xlinkHref="#icon-skype"
                />
              </svg>
            </a>
          </li>
          <li
            className="socials-item"
          >
            <a
              aria-label="vsco"
              className="socials__link"
              href="https://www.vsco.com/"
            >
              <svg
                aria-hidden="true"
                className="socials__icon"
                height="24"
                width="24"
              >
                <use
                  xlinkHref="#icon-vsco"
                />
              </svg>
            </a>
          </li>
          <li
            className="socials-item"
          >
            <a
              aria-label="pinterest"
              className="socials__link"
              href="https://www.pinterest.com/"
            >
              <svg
                aria-hidden="true"
                className="socials__icon"
                height="24"
                width="24"
              >
                <use
                  xlinkHref="#icon-pinterest"
                />
              </svg>
            </a>
          </li>
        </ul>
      </div>
      <section
        className="footer__nav-section footer__nav-section--info"
      >
        <h2
          className="footer__nav-title"
        >
          О нас
        </h2>
        <p
          className="footer__nav-content footer__nav-content--font-secondary"
        >
          Магазин гитар, музыкальных инструментов и гитарная мастерская 
          <br />
           в Санкт-Петербурге.
          <br />
          <br />
          Все инструменты проверены, отстроены 
          <br />
           и доведены до идеала!
        </p>
      </section>
      <section
        className="footer__nav-section footer__nav-section--links"
      >
        <h2
          className="footer__nav-title"
        >
          Информация
        </h2>
        <ul
          className="footer__nav-list"
        >
          <li
            className="footer__nav-list-item"
          >
            <a
              className="link"
              href="#top"
            >
              Где купить?
            </a>
          </li>
          <li
            className="footer__nav-list-item"
          >
            <a
              className="link"
              href="#top"
            >
              Блог
            </a>
          </li>
          <li
            className="footer__nav-list-item"
          >
            <a
              className="link"
              href="#top"
            >
              Вопрос - ответ
            </a>
          </li>
          <li
            className="footer__nav-list-item"
          >
            <a
              className="link"
              href="#top"
            >
              Возврат
            </a>
          </li>
          <li
            className="footer__nav-list-item"
          >
            <a
              className="link"
              href="#top"
            >
              Сервис-центры
            </a>
          </li>
        </ul>
      </section>
      <section
        className="footer__nav-section footer__nav-section--contacts"
      >
        <h2
          className="footer__nav-title"
        >
          Контакты
        </h2>
        <p
          className="footer__nav-content"
        >
          г. Санкт-Петербург,
          <br />
           м. Невский проспект, 
          <br />
          ул. Казанская 6.
        </p>
        <div
          className="footer__nav-content"
        >
          <svg
            aria-hidden="true"
            className="footer__icon"
            height="8"
            width="8"
          >
            <use
              xlinkHref="#icon-phone"
            />
          </svg>
          <a
            className="link"
            href="tel:88125005050"
          >
             8-812-500-50-50
          </a>
        </div>
        <p
          className="footer__nav-content"
        >
          Режим работы:
          <br />
          <span
            className="footer__span"
          >
            <svg
              aria-hidden="true"
              className="footer__icon"
              height="13"
              width="13"
            >
              <use
                xlinkHref="#icon-clock"
              />
            </svg>
            <span>
               с 11:00 до 20:00
            </span>
            <span>
              без выходных
            </span>
          </span>
        </p>
      </section>
    </div>
  </footer>
</div>
`;
